lib/binary-build.js

```diff
@@ -2,35 +2,41 @@
 
 const watchBuildLog = require('./build-watcher');
 
-// Start the build process
-module.exports = (config, archiveLocation) => {
+function wait (timeout) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve()
+    }, timeout)
+  })
+}
+
+module.exports = async (config, archiveLocation) => {
   // Upload the .tar file
-  return config.openshiftRestClient.buildconfigs.instantiateBinary(config.buildName, {dockerArchive: archiveLocation}).then((response) => {
-    console.log('Binary Upload Complete');
-
-    // Trigger watch of build
-    const watchPromise = watchBuildLog(config, response.metadata.name);
-    const timerPromise = new Promise((resolve, reject) => {
-      console.log('Waiting for build to finish');
-      const internvalId = setInterval(() => {
-        // find the current build
-        config.openshiftRestClient.builds.find(response.metadata.name).then((buildStatus) => {
-          if (buildStatus.status.phase === 'Complete') {
-            clearInterval(internvalId);
-            console.log('Build finished');
-            return resolve(buildStatus);
-          }
-
-          if (buildStatus.status.phase === 'Failed') {
-            clearInterval(internvalId);
-            console.log('Build Failed with message:', buildStatus.status.message);
-            return reject(new Error(buildStatus.status.message));
-          }
-        });
-      }, 2000);
-    });
-    return Promise.all([timerPromise, watchPromise]).then((results) => {
-      return results[0];
-    });
-  });
-};
+  const response = await config.openshiftRestClient
+    .buildconfigs
+    .instantiateBinary(config.buildName, { dockerArchive: archiveLocation });
+
+  console.log('Binary upload complete');
+
+  // Trigger watch of build
+  await watchBuildLog(config, response.metadata.name);
```

Comment:
I noticed that the watching of the logs and checking the build status are now sequential instead of happening in parallel.

although it might not matter